resources/sdk/clisdkclient/scripts/preprocess-swagger.js

```diff
@@ -1,175 +1,186 @@
+				}
 				if (duplicateCommandMappings.get(`${commandName}_${supercommand}`)) {
-					commandName = `${commandName}_${supercommand}`
+					commandName = `${commandName}_${supercommand}`;
 				}
 			}
-			value.tags = [commandName]
+			value.tags = [commandName];
 		}
 
 		// Override operationId if possible
 		for (const method of Object.keys(includedSwaggerPathObjects[path])) {
-			if (!Object.keys(resourceDefinitions[path]).includes(method)) continue
+			if (!Object.keys(resourceDefinitions[path]).includes(method)) continue;
 			if (resourceDefinitions[path][method].name !== undefined) {
-				includedSwaggerPathObjects[path][method].operationId = `SWAGGER_OVERRIDE_${resourceDefinitions[path][method].name}`
+				includedSwaggerPathObjects[path][method].operationId = `SWAGGER_OVERRIDE_${resourceDefinitions[path][method].name}`;
 			}
 		}
 
 		// Exclude HTTP methods if possible
-		let includedMethods = []
-		const httpMethods = ["get", "delete", "put", "patch", "post"]
+		let includedMethods = [];
+		const httpMethods = ['get', 'delete', 'put', 'patch', 'post'];
 		for (const httpMethod of Object.keys(resourceDefinitions[path])) {
-			if (!httpMethods.includes(httpMethod)) continue
-			includedMethods.push(httpMethod)
+			if (!httpMethods.includes(httpMethod)) continue;
+			includedMethods.push(httpMethod);
 		}
 
-		paths[path] = {}
+		paths[path] = {};
 		if (includedMethods.length > 0) {
 			for (const method of includedMethods) {
-				paths[path][method] = includedSwaggerPathObjects[path][method]
+				paths[path][method] = includedSwaggerPathObjects[path][method];
 			}
 		} else {
-			paths[path] = includedSwaggerPathObjects[path]
+			paths[path] = includedSwaggerPathObjects[path];
 		}
 	}
 
-	return paths
+	return paths;
 }
 
 function findDuplicateCommandMappings(superCommands, includedSwaggerPathObjects, resourceDefinitions) {
-	let duplicateCommandMappings = new Map()
+	let duplicateCommandMappings = new Map();
 	for (const path of Object.keys(includedSwaggerPathObjects)) {
 		for (let value of Object.values(includedSwaggerPathObjects[path])) {
-			let commandName = resourceDefinitions[path].name || value.tags[0]
-			commandName = commandName.toLowerCase().replace(" ", "")
+			let commandName = resourceDefinitions[path].name || value.tags[0];
+			commandName = commandName.toLowerCase().replace(' ', '');
 
-			const supercommand = resourceDefinitions[path].supercommand
+			let supercommandList = resourceDefinitions[path].supercommand;
 			// Only need to work on duplicate subcommands, no one would try to create 2 `gc users` commands for example
-			if (supercommand) {
-				const existingCommandName = duplicateCommandMappings.get(commandName)
+			if (supercommandList) {
+				const existingCommandName = duplicateCommandMappings.get(commandName);
+				let supercommandSplit = supercommandList.split('.');
+				let supercommand = supercommandSplit.pop();
+				for (var i = supercommandSplit.length - 1; i >= 0; i--) {
```

Comment:
When possible use a foreach loop.  Using length and booleans are susceptible to post errors

Reply:
So for this, and also the next comment, it's traversing backwards, so should I use list.slice().reverse().foreach => {}? Seems like overkill? But it that is the use case, I'll change it.